src/containers/Cluster/ClusterDashboard/ClusterDashboard.tsx

```diff
@@ -0,0 +1,137 @@
+import {Flex, Text} from '@gravity-ui/uikit';
+
+import {Tags} from '../../../components/Tags';
+import type {ClusterGroupsStats} from '../../../store/reducers/cluster/types';
+import {isClusterInfoV2} from '../../../types/api/cluster';
+import type {TClusterInfo} from '../../../types/api/cluster';
+import {valueIsDefined} from '../../../utils';
+import {formatNumber} from '../../../utils/dataFormatters/dataFormatters';
+import i18n from '../i18n';
+
+import {ClusterMetricsCard, ClusterMetricsCardSkeleton} from './components/ClusterMetricsCard';
+import {ClusterMetricsCores} from './components/ClusterMetricsCores';
+import {ClusterMetricsMemory} from './components/ClusterMetricsMemory';
+import {ClusterMetricsStorage} from './components/ClusterMetricsStorage';
+import {b} from './shared';
+import {
+    getDCInfo,
+    getNodesRolesInfo,
+    getStorageGroupStats,
+    getTotalStorageGroupsUsed,
+} from './utils';
+
+import './ClusterDashboard.scss';
+
+interface AmountProps {
+    value?: number | string;
+}
+
+function Amount({value}: AmountProps) {
+    if (!valueIsDefined(value)) {
+        return null;
+    }
+    return (
+        <Text variant="subheader-3" color="secondary">
+            {formatNumber(value)}
+        </Text>
+    );
+}
+
+interface ClusterDashboardProps {
+    cluster: TClusterInfo;
+    groupStats?: ClusterGroupsStats;
+    loading?: boolean;
+}
+
+export function ClusterDashboard({cluster, groupStats = {}, loading}: ClusterDashboardProps) {
+    const getDoughnuts = () => {
+        if (loading) {
+            return Array.from('123').map((el) => <ClusterMetricsCardSkeleton key={el} />);
+        }
+        const metricsCards = [];
+        if (isClusterInfoV2(cluster)) {
+            const {CoresUsed, NumberOfCpus} = cluster;
+            if (valueIsDefined(CoresUsed) && valueIsDefined(NumberOfCpus)) {
+                metricsCards.push(
+                    <ClusterMetricsCores value={CoresUsed} capacity={NumberOfCpus} key="cores" />,
+                );
+            }
+        }
+        const {StorageTotal, StorageUsed} = cluster;
+        if (valueIsDefined(StorageTotal) && valueIsDefined(StorageUsed)) {
+            metricsCards.push(
+                <ClusterMetricsStorage key="storage" value={StorageUsed} capacity={StorageTotal} />,
+            );
+        }
+        const {MemoryTotal, MemoryUsed} = cluster;
+        if (valueIsDefined(MemoryTotal) && valueIsDefined(MemoryUsed)) {
+            metricsCards.push(
+                <ClusterMetricsMemory key="memory" value={MemoryUsed} capacity={MemoryTotal} />,
+            );
+        }
+        return metricsCards;
+    };
+
+    const getCards = () => {
```

Comment:
imho It would be better to make it distinct component to make the main component more readable